jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanPropertyIntrospector.java

```diff
@@ -1,40 +1,42 @@
 package com.fasterxml.jackson.jr.ob.impl;
 
-import java.lang.reflect.*;
+import com.fasterxml.jackson.jr.ob.impl.POJODefinition.Prop;
+import com.fasterxml.jackson.jr.ob.impl.POJODefinition.PropBuilder;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.TreeMap;
 
-import com.fasterxml.jackson.jr.ob.JSON;
-import com.fasterxml.jackson.jr.ob.impl.POJODefinition.Prop;
-import com.fasterxml.jackson.jr.ob.impl.POJODefinition.PropBuilder;
+import static com.fasterxml.jackson.jr.ob.JSON.Feature.INCLUDE_STATIC_FIELDS;
+import static com.fasterxml.jackson.jr.ob.JSON.Feature.USE_FIELD_MATCHING_GETTERS;
+import static java.lang.Character.isLowerCase;
+import static java.lang.Character.toLowerCase;
+import static java.lang.reflect.Modifier.*;
 
 /**
  * Helper class that jackson-jr uses by default to introspect POJO properties
  * (represented as {@link POJODefinition}) to build general POJO readers
  * (deserializers) and writers (serializers).
- *<p>
+ * <p>
  * Note that most of the usage is via {@link ValueReaderLocator} and
  * {@link ValueWriterLocator}
  *
  * @since 2.11
  */
-public class BeanPropertyIntrospector
-{
-    protected final static Prop[] NO_PROPS = new Prop[0];
-
-    private final static BeanPropertyIntrospector INSTANCE = new BeanPropertyIntrospector();
-
-    public BeanPropertyIntrospector() { }
-
-    public static BeanPropertyIntrospector instance() { return INSTANCE; }
+public final class BeanPropertyIntrospector {
+    private BeanPropertyIntrospector() {
+    }
 
-    public POJODefinition pojoDefinitionForDeserialization(JSONReader r, Class<?> pojoType) {
-        return _introspectDefinition(pojoType, false, r.features());
+    public static POJODefinition pojoDefinitionForDeserialization(JSONReader r, Class<?> pojoType) {
```

Comment:
can't change a public method to be static in a patch

Reply:
I get your concern, this change can have unforeseen side-effects.
Currently this is just a proof of concept, to simplify and cleanup this class. 

we can take items from here individually (if not all). or not use this at all (whatever is feasible)